nifi-nar-bundles/nifi-spring-bundle/nifi-spring-processors/src/main/java/org/apache/nifi/spring/SpringContextProcessor.java

```
@@ -0,0 +1,392 @@
+ * Spring) - depends on existence of both "fromNiFi" and "toNiFi" channels in
+ * the Spring Application Context</li>
+ * </ul>
+ *
+ * </p>
+ * <p>
+ * To create an instance of the ApplicationConetxt this processor requires user
+ * to provide configuration file path and the path to the resources that needs
+ * to be added to the classpath of ApplicationContext. This essentially allows
+ * user to package their Spring Application any way they want as long as
+ * everything it requires is available on the classpath.
+ * </p>
+ * <p>
+ * Data exchange between Spring and NiFi relies on simple mechanism which is
+ * exposed via {@link SpringDataExchanger}; {@link FlowFile}s's content is
+ * converted to primitive representation that can be easily wrapped in Spring
+ * {@link Message}. The requirement imposed by this Processor is to send/receive
+ * {@link Message} with <i>payload</i> of type <i>byte[]</i> and headers of type
+ * <i>Map&lt;String, Object&gt;</i>. This is primarily for simplicity and type
+ * safety. Converters and Transformers could be used by either side to change
+ * representation of the content that is being exchanged between NiFi and
+ * Spring.
+ */
+@TriggerWhenEmpty
+@Tags({ "Spring", "Message", "Get", "Put", "Integration" })
+@CapabilityDescription("A Processor that supports sending and receiving data from application defined in "
+        + "Spring Application Context via predefined in/out MessageChannels.")
+public class SpringContextProcessor extends AbstractProcessor {
+    private final Logger logger = LoggerFactory.getLogger(SpringContextProcessor.class);
+
+    public static final PropertyDescriptor CTX_CONFIG_NAME = new PropertyDescriptor.Builder()
+            .name("Application Context config path")
+            .description("The path to the Spring Application Context configuration file relative to the classpath")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
```

Even if we were to switch the order of the properties, unless it could be enforced there is no joy. So I say leave it and let it fail at the start.